server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -652,6 +652,11 @@ private void innerExecute(int slot, IndexRequest indexRequest, Pipeline pipeline
                     indexRequest.setIfPrimaryTerm(((Number) metadataMap.get(IngestDocument.Metadata.IF_PRIMARY_TERM)).longValue());
                 }
                 indexRequest.source(ingestDocument.getSourceAndMetadata(), indexRequest.getContentType());
+                if (metadataMap.get(IngestDocument.Metadata.DYNAMIC_TEMPLATES) != null) {
+                    Map<String, String> mergedDynamicTemplates = new HashMap<>(indexRequest.getDynamicTemplates());
+                    mergedDynamicTemplates.putAll((Map<String, String>) metadataMap.get(IngestDocument.Metadata.DYNAMIC_TEMPLATES));
+                    indexRequest.setDynamicTemplates(mergedDynamicTemplates);
```

Comment:
can we add docs?  As a user, without looking at the code, I would wonder if I end up overwriting what has been set by the bulk request. Also, whether I have access to the templates that are already set or I can only provide new ones.

Reply:
++, I added a doc in https://github.com/elastic/elasticsearch/pull/71716/commits/41f08b78dbfeb95aea616cc377556a34ba093055.